kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/LocalRule.java

```diff
@@ -407,61 +407,34 @@ public String getPermittedMAC() {
      */
     @Override
     public String toString() {
-        String interfaceString = null;
+        StringBuilder localRuleSB = new StringBuilder(A_INPUT_KURA_P).append(this.protocol);
+        
+        this.permittedNetworkString.ifPresent(permittedNetwork -> {
+            localRuleSB.append(" -s ").append(permittedNetwork);
+        });
+        
         if (this.permittedInterfaceName.isPresent()) {
-            interfaceString = new StringBuilder().append(" -i ").append(this.permittedInterfaceName.get()).toString();
+            localRuleSB.append(" -i ").append(this.permittedInterfaceName.get()).toString();
         } else if (this.unpermittedInterfaceName.isPresent()) {
-            interfaceString = new StringBuilder().append(" ! -i ").append(this.unpermittedInterfaceName.get())
-                    .toString();
+            localRuleSB.append(" ! -i ").append(this.unpermittedInterfaceName.get()).toString();
         }
 
+        this.permittedMAC.ifPresent(permittedMAC -> {
+            localRuleSB.append(M_MAC_MAC_SOURCE).append(permittedMAC);
+        });
```

Comment:
Also... `permittedMAC` hides a field declared on line 42. I would rename it just to be on the safe side